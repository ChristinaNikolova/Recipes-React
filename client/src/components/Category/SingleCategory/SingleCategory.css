.custom-card-category {
    width: 32%;
    margin-bottom: 15px;
}

.custom-card-category h3.card-header {
    background-color: #54BAB9;
    font-weight: 400;
    letter-spacing: 1px;
    color: #F7ECDE;
    text-align: center;
}

.category-img {
    width: 100%;
    height: 300px;
    object-fit: cover;
    border: 0.1rem hidden;
    padding: 10px;
    background-color: white;
}

.category--recipe-count {
    border-top: 1px solid lightgrey;
    border-bottom: 1px solid lightgrey;
    padding: 18px 15px;
    margin-bottom: unset;
    font-size: 17px;
    letter-spacing: .8px;
}

.custom-card-category .fa-book {
    color: #54BAB9;
}

.card-body .custom-btn {
    width: 50%;
    font-size: 18px;
    margin-bottom: 10px;
}