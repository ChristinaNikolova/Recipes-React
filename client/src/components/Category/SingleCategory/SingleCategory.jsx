import { Link } from 'react-router-dom';

import './SingleCategory.css';

function SingleCategory({ id, name, picture, recipesCount }) {
    return (
        <div className="custom-card-category">
            <div className="card">
                <h3 className="card-header cursive-font-style">{name}</h3>
                <img className="category-img" src={picture} alt="category-pic"></img>
                <ul className='category--recipe-count'>
                    <li>
                        <i className="fas fa-book mr-1"></i>
                        Recipes in category: {recipesCount}
                    </li>
                </ul>
                <div className="card-body">
                    <Link to={`/recipes/current-category/${id}`}>
                        <button className="custom-btn">See recipes</button>
                    </Link>
                </div>
            </div>
        </div>
    );
}

export default SingleCategory;