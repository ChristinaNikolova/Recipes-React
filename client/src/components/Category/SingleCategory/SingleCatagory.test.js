import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SingleCategory from './SingleCategory';

it('should render SingleCategory component', () => {
    const wrapper = shallow(<SingleCategory />);
    expect(toJson(wrapper)).toMatchSnapshot();
});