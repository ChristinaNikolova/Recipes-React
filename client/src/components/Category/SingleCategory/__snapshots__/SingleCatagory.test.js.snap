// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SingleCategory component 1`] = `
<div
  className="col-lg-4 custom-card-category"
>
  <div
    className="card mb-2"
  >
    <h3
      className="card-header category-title cursive-font-style"
    />
    <img
      alt="category-pic"
      className="img-category"
    />
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <i
          className="fas fa-book mr-1"
        />
        Recipes in category: 
      </li>
    </ul>
    <div
      className="card-body"
    >
      <Link
        to="/recipes/current-category/undefined"
      >
        <button
          className="btn btn-secondary button-category"
        >
          See recipes
        </button>
      </Link>
    </div>
  </div>
</div>
`;
